src/quacc/runners/ase.py

```diff
@@ -240,6 +246,132 @@ def run_opt(
+    """
+
+    # Copy atoms so we don't modify it in-place
+    atoms = copy_atoms(atoms)
+
+    # Perform staging operations
+    tmpdir, job_results_dir = calc_setup(atoms, copy_files=copy_files)
+
+    # Set defaults
+    dynamics_kwargs = recursive_dict_merge(
+        {"logfile": "-" if SETTINGS.DEBUG else tmpdir / "dyn.log"}, dynamics_kwargs
+    )
+
+    # Check if trajectory kwarg is specified
+    if "trajectory" in dynamics_kwargs:
+        msg = "Quacc does not support setting the `trajectory` kwarg."
+        raise ValueError(msg)
+
+    if "temperature" in dynamics_kwargs or "temp" in dynamics_kwargs:
+        LOGGER.warning(
+            r"The `temperature`\`temp` kwargs are ASE deprecated and will"
+            "be interpreted as `temperature_K` in Quacc."
+        )
+        dynamics_kwargs["temperature_K"] = dynamics_kwargs.pop(
+            "temperature", None
+        ) or dynamics_kwargs.pop("temp", None)
+
+    if "pressure" in dynamics_kwargs:
+        LOGGER.warning(
+            "The `pressure` kwarg is ASE deprecated and will"
+            "be interpreted as `pressure_au` in Quacc."
+        )
+        dynamics_kwargs["pressure_au"] = dynamics_kwargs.pop("pressure")
+
+    if "compressibility" in dynamics_kwargs:
+        LOGGER.warning(
+            "The `compressibility` kwarg is ASE deprecated and will"
+            "be interpreted as `compressibility_au` in Quacc."
+        )
+        dynamics_kwargs["compressibility_au"] = dynamics_kwargs.pop("compressibility")
+
+    if "dt" in dynamics_kwargs:
+        LOGGER.warning(
+            "The `dt` kwarg is ASE deprecated and will"
+            "be interpreted as `timestep` in Quacc."
+        )
+        dynamics_kwargs["timestep"] = dynamics_kwargs.pop("dt")
+
+    # Convert units so that Quacc ALWAYS use the same units, for
+    # every MD classes.
+    # Users will have to always input time in fs, pressure in GPa
+    # temperature in K, and compressibility in 1/GPa.
+
+    if "ttime" in dynamics_kwargs:
+        dynamics_kwargs["ttime"] = dynamics_kwargs.pop("ttime") * fs
+
+    if "friction" in dynamics_kwargs:
+        dynamics_kwargs["friction"] = dynamics_kwargs.pop("friction") / fs
+
+    if "taut" in dynamics_kwargs:
+        dynamics_kwargs["taut"] = dynamics_kwargs.pop("taut") * fs
+
+    if "taup" in dynamics_kwargs:
+        dynamics_kwargs["taup"] = dynamics_kwargs.pop("taup") * fs
+
+    if "pfactor" in dynamics_kwargs:
+        dynamics_kwargs["pfactor"] = dynamics_kwargs.pop("pfactor") * GPa
+
+    if "compressibility_au" in dynamics_kwargs:
+        dynamics_kwargs["compressibility"] = (
+            dynamics_kwargs.pop("compressibility") / GPa
+        )
+
+    traj_filename = tmpdir / "opt.traj"
+    traj = Trajectory(traj_filename, "w", atoms=atoms)
+    dynamics_kwargs["trajectory"] = traj
+
+    # Run calculation
+    with traj, dynamics(atoms, timestep=timestep * fs, **dynamics_kwargs) as dyn:
+        dyn.run(steps=steps)
```

Comment:
Do you think it makes sense to add an option that would save the files from every step, e.g. for training ML models later? I did that recently for the optimization runner [here](https://github.com/Quantum-Accelerators/quacc/blob/a13538a7df5628117442c0745637bbe10e0c7aa5/src/quacc/runners/ase.py#L224-L238). I can add this to the MD run, but I am just curious about your opinion.

Reply:
Yes, I wanted to do something like that at some point. I think it's a good idea, although you might want to prepare a big scratch space for this 😅

That's also what I had in mind when thinking about the "optimizer patching". I presented it as a very general concept but in practice this could be used to do something like that. A "FileSave" patch that you would send to your ase_md_job. This patch would then have parameters to allow more flexibility like "save_interval" etc... Usually it would allow to save files based on some other criteria (geometric descriptors, ML auto-evaluation...)

To be fair for this particular action I think your approach is better.